bin/foreman-node

```diff
@@ -15,6 +15,10 @@ require 'net/https'
 require 'etc'
 require 'timeout'
 
+if SETTINGS[:filecache]
```

Comment:
If filecache is so much faster and "safe" to be used I have 2 thoughts:
1. why should we still use the salt-run call? should we remove salt-run completely?
2. if the salt-run call is not removed completely, should the filecache appraoch at least be the default?

Reply:
There may be cases where salt is configured to use external minon data cache as per 
https://docs.saltstack.com/en/latest/topics/cache/index.html

filecache will only work if the default, file-based minion data cache is used. 
salt-run will work for either, so keeping it makes sense.

I'll consider your other requested changes and take a look asap but probably not before next week.

Reply:
Then we need to document when this feature can be activated and which configuration on the salt master needs to be done.

Reply:
As for the documentation, I added a brief comment in the method within the code, but maybe this should be part of the actual docs?

Reply:
- yes, this would be good: https://github.com/theforeman/theforeman.org/tree/gh-pages/plugins/foreman_salt/13.0

Reply:
done